matrix_client/crypto/olm_device.py

```diff
@@ -50,6 +68,52 @@ def upload_identity_keys(self):
         self.one_time_key_counts = ret['one_time_key_counts']
         logger.info('Uploaded identity keys.')
 
+    def upload_one_time_keys(self, force_update=False):
+        """Uploads new one-time keys to the HS, if needed.
+
+        Args:
+            force_update (bool): Fetch the number of one-time keys currently on the HS
+                before uploading, even if we already know one. In most cases this should
+                not be necessary, as we get this value from sync responses.
+
+        Returns:
+            A dict containg the number of new keys that were uploaded for each key type
+                (signed_curve25519 or curve25519). The format is
+                ``<key_type>: <uploaded_number>``. If no keys of a given type have been
+                uploaded, the corresponding key will not be present. Consequently, an
+                empty dict indicates that no keys were uploaded.
+        """
+        if not self.one_time_key_counts or force_update:
+            self.one_time_key_counts = self.api.upload_keys()['one_time_key_counts']
+
+        keys_uploaded = {}
+        for key_type, target_number in self.otk_target_counts.items():
+            # The key_type key will not be present before our first upload
+            num_keys = self.one_time_key_counts.get(key_type, 0)
+            num_to_create = max(target_number - num_keys, 0)
+            if not num_to_create:
+                continue
+            keys_uploaded[key_type] = num_to_create
+        self.olm_account.generate_one_time_keys(sum(keys_uploaded.values()))
```

Comment:
Given that you explicitly name the keys as curve25519 in `self.otk_target_counts`, doesn't this do the wrong thing? Having "curve25519" included in name of dictionary keys implies there could be other algos in dictionary in the future, and this call shouldn't care about that.